lib/bulking.ts

```diff
@@ -0,0 +1,66 @@
+import { BulkIndexOptions, BulkInstruction, BulkOptions, BulkUnIndexOptions, MongoosasticDocument, MongoosasticModel } from 'types'
+
+let bulkBuffer: BulkInstruction[] = []
+let bulkTimeout: NodeJS.Timeout | undefined
+
+function clearBulkTimeout() {
+	clearTimeout(bulkTimeout as NodeJS.Timeout)
+	bulkTimeout = undefined
+}
+
+export async function bulkAdd(opts: BulkIndexOptions): Promise<void> {
+	const instruction = [{
+		index: {
+			_index: opts.index,
+			_id: opts.id,
+		}
+	}, opts.body]
+	
+	await bulkIndex(opts.model, instruction, opts.bulk as BulkOptions)
```

Comment:
I would suggest return the operation instead of void

Reply:
this function is used in "bulkAdd" & "bulkDelete" (to add the bulk instruction to the bulk buffer, and flush it later) which are called in "index" & "unIndex" methods. 
Those last two methods return the indexed document, whatever the bulk response (We handle the bulking errors using an event emitter `bulkError`). So i did not see the need to return the operation.